### PR TITLE
release 0.0.2

### PR DESCRIPTION
**Pull request by luminerrr**
created several endpoints

please approve suhu 🙏🙏🙏deadline jam 5 🙏🙏🙏

### CODE REVIEW

File: src/main/java/com/example/suka_bapak/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        try {
+            bookService.updateBook(bookId, createBookRequest);
+            return new ResponseEntity<>(Map.of("message", "Book details updated successfully."), HttpStatus.OK);
+        } catch (ValidationException e) {
+            return new ResponseEntity<>(Map.of("error", e.getMessage()), HttpStatus.BAD_REQUEST);
+        } catch (RuntimeException e) {
+            return new ResponseEntity<>(Map.of("error", "Book not found."), HttpStatus.NOT_FOUND);
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (luminerrr) [reply src/main/java/com/example/suka_bapak/controller/BookController.java:76]: ditto apa mas

[the remark this answers is not in the document]
Comment (khalidalhabibie) [reply src/main/java/com/example/suka_bapak/controller/BookController.java:76]: di masukkan juga ke dalam service

[the remark this answers is not in the document]
Comment (khalidalhabibie) [reply src/main/java/com/example/suka_bapak/controller/BookController.java:76]: jadi 

```
public ResponseEntity<?> updateBook(
            @PathVariable("book_id") Long bookId,
            @RequestBody CreateBookRequest createBookRequest) {
 
             return bookService.updateBook(bookId, createBookRequest);
        
    }


```

try catch ke dalam service aja

File: src/main/java/com/example/suka_bapak/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            return new ResponseEntity<>(Map.of(
+                    "id", newBook.getId(),
+                    "title", newBook.getTitle(),
+                    "available_copies", newBook.getAvailable_copies()
+            ), HttpStatus.CREATED);
+        } catch (ValidationException e) {
+            return new ResponseEntity<>(Map.of("error", e.getMessage()), HttpStatus.BAD_REQUEST);
+        }
+    }
+//    public ResponseEntity<BookEntity> createBook(@RequestBody CreateBookRequest book) {
```
[the remark this answers is not in the document]
Comment (luminerrr) [reply src/main/java/com/example/suka_bapak/controller/BookController.java]: done

File: src/main/java/com/example/suka_bapak/controller/PatronController.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            @RequestBody CreatePatronRequest createPatronRequest) {
+        try {
+            patronService.updatePatron(id, createPatronRequest);
+            return new ResponseEntity<>(Map.of("message", "Patron details updated successfully."), HttpStatus.OK);
+        } catch (ValidationException e) {
+            return new ResponseEntity<>(Map.of("error", e.getMessage()), HttpStatus.BAD_REQUEST);
+        } catch (RuntimeException e) {
+            return new ResponseEntity<>(Map.of("error", "Patron not found."), HttpStatus.NOT_FOUND);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (luminerrr) [reply src/main/java/com/example/suka_bapak/controller/PatronController.java]: done

File: src/main/java/com/example/suka_bapak/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+//        BookEntity savedBook = bookService.createBook(book);
+//        return ResponseEntity.ok(savedBook);
+//    }
```
[the remark this answers is not in the document]
Comment (luminerrr) [reply src/main/java/com/example/suka_bapak/controller/BookController.java]: done

File: src/main/java/com/example/suka_bapak/dto/response/patrons/GetPatronDto.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+
```
[the remark this answers is not in the document]
Comment (luminerrr) [reply src/main/java/com/example/suka_bapak/dto/response/patrons/GetPatronDto.java:10]: done

File: src/main/java/com/example/suka_bapak/controller/BookController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (luminerrr) [reply src/main/java/com/example/suka_bapak/controller/BookController.java]: i see, done ya mas

**Review by khalidalhabibie — APPROVED** [review]
LGTM!

**Pull request merged**